test/functional/feature_anchors.py

```diff
@@ -63,14 +63,21 @@ def run_test(self):
         self.log.info("Check the addresses in anchors.dat")
 
         with open(node_anchors_path, "rb") as file_handler:
-            anchors = file_handler.read().hex()
+            anchors = file_handler.read()
 
+        anchors_hex = anchors.hex()
         for port in block_relay_nodes_port:
             ip_port = ip + port
-            assert ip_port in anchors
+            assert ip_port in anchors_hex
         for port in inbound_nodes_port:
             ip_port = ip + port
-            assert ip_port not in anchors
+            assert ip_port not in anchors_hex
+
+        self.log.info("Perturb anchors.dat to test it doesn't throw an error during initialization")
+        with open(node_anchors_path, "wb") as out_file_handler:
+            tweaked_contents = bytearray(anchors)
+            tweaked_contents[10:20] = b'1' * 25
+            out_file_handler.write(bytes(tweaked_contents))
```

Comment:
Maybe writing a single byte b'a' or so achieves the same?

Reply:
Yes, it achieves the same, just tested here manually btw... Do you think it's best to change it to modify a single byte ? e.g.:
```diff
diff --git a/test/functional/feature_anchors.py b/test/functional/feature_anchors.py
index aa80b9f1d..322b6b177 100755
--- a/test/functional/feature_anchors.py
+++ b/test/functional/feature_anchors.py
@@ -76,7 +76,7 @@ class AnchorsTest(BitcoinTestFramework):
         self.log.info("Perturb anchors.dat to test it doesn't throw an error during initialization")
         with open(node_anchors_path, "wb") as out_file_handler:
             tweaked_contents = bytearray(anchors)
-            tweaked_contents[10:20] = b'1' * 25
+            tweaked_contents[35:35] = b'1'
             out_file_handler.write(bytes(tweaked_contents))
```